x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs.tsx

```diff
@@ -5,22 +5,40 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
+
+import { useValues, useActions } from 'kea';
 
 import { EuiPageHeader, EuiTitle, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
 
 import { FlashMessages } from '../../../shared/flash_messages';
 import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
 import { BreadcrumbTrail } from '../../../shared/kibana_chrome/generate_breadcrumbs';
+import { Loading } from '../../../shared/loading';
 
 import { LogRetentionCallout, LogRetentionTooltip, LogRetentionOptions } from '../log_retention';
 
 import { API_LOGS_TITLE, RECENT_API_EVENTS } from './constants';
 
+import { ApiLogsLogic } from './';
+
 interface Props {
   engineBreadcrumb: BreadcrumbTrail;
 }
 export const ApiLogs: React.FC<Props> = ({ engineBreadcrumb }) => {
+  const { dataLoading, apiLogs, meta } = useValues(ApiLogsLogic);
+  const { fetchApiLogs, pollForApiLogs } = useActions(ApiLogsLogic);
+
+  useEffect(() => {
+    fetchApiLogs();
+  }, [meta.page.current]);
+
+  useEffect(() => {
+    pollForApiLogs();
+  }, []);
```

Comment:
^ I hope with this useEffect example it makes more sense why I split out the concept of an "immediate" fetch and a "poll" fetch in the logic! Also for even more context, check out the different requirements for the Engine Overview:

https://github.com/elastic/kibana/blob/2250d8228d4466dbf5df249583033a57d04d1f9c/x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_overview/components/recent_api_logs.tsx#L24-L27

The table there **does not** have pagination, so it doesn't need an extra effect and only calls `fetchApiLogs` once on load.